torch/csrc/jit/frontend/function_schema_parser.cpp

```diff
@@ -219,18 +220,14 @@ struct SchemaParser {
       case TK_IDENT: {
         auto tok = L.next();
         auto text = tok.text();
-        if ("float" == text) {
-          return static_cast<int64_t>(at::kFloat);
-        } else if ("complex" == text) {
-          return static_cast<int64_t>(at::kComplexFloat);
-        } else if ("long" == text) {
-          return static_cast<int64_t>(at::kLong);
-        } else if ("strided" == text) {
+        if ("strided" == text) {
           return static_cast<int64_t>(at::kStrided);
         } else if ("Mean" == text) {
           return static_cast<int64_t>(at::Reduction::Mean);
         } else if ("contiguous_format" == text) {
           return static_cast<int64_t>(c10::MemoryFormat::Contiguous);
+        } else if (str2dtype.count(text) > 0) {
```

Comment:
Could you clarify the implication of this change? I'm not familiar with when this code is being used.

Reply:
This function gets called during construction of a FunctionSchema (https://www.internalfb.com/code/fbsource/[aca8f430d0152e7ffe6b24aedb8ee29808a16e3f]/fbcode/caffe2/torch/csrc/jit/frontend/function_schema_parser.cpp?lines=200%2C202%2C285%2C314%2C327): it is responsible for parsing the default values in the schema and turning them into an IValue. The implication of the change is now we support specifying the default value to be any of the keys in the str2dtype mapping (like "float64", "complex128", etc).

Reply:
Ok!
I guess none of these will be ambiguous!

Reply:
There's a bug here though, we only want to parse this as a dtype if we're sure the type is a dtype, otherwise weird things like "int x = complex128" will work... so I'll try to fix that. The bug is pre-existing though

Reply:
Yeah I guess anything that gets silently converted to int on the way down can have anything as it's default haha